api/src/models/ImageFile.ts

```diff
@@ -65,34 +65,52 @@ class ImageFile {
         return dir + "/" + filename + "/" + size + "/" + filename + "." + extension.toLowerCase();
     }
 
-    ocr() {
+    async ocr() {
         //TODO: update the following (derivativepath requires size now)
         var txt = this.derivativePath('OCR-DIRTY', 'txt'); 
         let fs = require('fs');
-        var tesseract = require("node-tesseract-ocr");
-        if (fs.existsSync(txt)){
+        let { exec } = require("child_process");
+        let deriv = await this.ocrDerivative();
+        if (!fs.existsSync(txt)){
             var path = this.basename(txt);
+            let ts_cmd = this.config().tesseractPath() + " " + deriv + " " + txt.slice(0, -4) + " " + this.ocrProperties();
+            exec(ts_cmd, (error, stdout, stderr) => {
+                if (error) {
+                    throw `error: ${error.message}`;
+                    return;
```

Comment:
I think you might be able to refine this a little bit...

1.) I don't think it's necessary to `return` after a `throw`, because the `throw` will already cause you to exit the function.
2.) I don't know if you want to throw an error just because there is content in the stderr stream; it's possible that Tesseract might throw data to stderr even when successful (or not... I can't remember). The error parameter is the main thing to be worried about, since if it is non-null, there is definitely a problem. Maybe the approach should be to include the stderr output in the exception thrown when error is non-null, and to console.log it for informational purposes under other circumstances.